### ISSUE TITLE
Padded Output Width vs Output Width

### ISSUE THREAD

**Issue by jc211** [NVlabs/tiny-cuda-nn#266]
All the encodings use:
```cpp

uint32_t padded_output_width() const override {
return m_n_output_dims + m_n_to_pad;
}

uint32_t output_width() const override {
return padded_output_width();
}

```

Why isn't output_width returning the non padded width?

**Comment by Tom94** [comment]
Hi, this is an unfortunate clash of naming. In tcnn, there are two kinds of output padding:
1. padding that is meant to be ignored by downstream models (and which is automatically trimmed away when calling `DifferentiableObject::inference(...)`. The neural networks do this. Hence, `Network` objects have a distinction between `output_width()` and `padded_output_width()`.
2. padding that is meant to be consumed by downstream models. Encodings do this, because there is no easy way downstream matrix multiplies to ignore some of their input dimensions. In this case, the encoding is meant to look from the outside as if its padding is part of its intended output, hence `output_width()` is the same as `padded_output_width()`.

**Comment by jc211** [comment]
Hey Tom, 

Thanks for your feedback. 

Does this mean that when a frequency encoding is used, the network is also consuming whatever values exist within the padded output?

Also, does this not also mean that there are no getters for m_n_output_dims within encoding? For a usecase where I'd like to concatenate 2 encodings together whilst ignoring their padded values how would one what the m_n_ouput_dims is?

**Comment by Tom94** [comment]
Yes, that's correct! For you use case of concatenating two encodings, you should already be able to use `CompositeEncoding`, no?
